### PR TITLE
🛠️ Add SQS to AP Development

### PR DESCRIPTION
**Pull request by Gary-H9**
The branch for this is a WIP. Commit messages will be squashed prior to merge.

### CODE REVIEW

**Comment by Gary-H9** [comment]
Further work needs to be completed on the IAM Policy permissions.

**Review by mitchdawson1982 — DISMISSED** [review]
LGTM

File: terraform/aws/analytical-platform-development/cluster/iam-policies.tf [hunk 1]
```diff
@@ -472,12 +472,13 @@ data "aws_iam_policy_document" "control_panel_api" {
...
     effect = "Allow"
     actions = [
+      "sqs:GetQueueAttributes",
       "sqs:GetQueueUrl",
       "sqs:DeleteMessage",
       "sqs:ReceiveMessage",
       "sqs:SendMessage"
     ]
-    resources = ["arn:aws:sqs::${var.account_ids["analytical-platform-development"]}:*"]
+    resources = ["arn:aws:sqs:eu-west-2:${var.account_ids["analytical-platform-development"]}:*"]
```
Comment (julialawrence) [inline terraform/aws/analytical-platform-development/cluster/iam-policies.tf]: If this is running in eu-west-2, I wouldn't hardcode the region but use a datasource to grab it. 
Example here:
https://stackoverflow.com/questions/51619602/terraform-lookup-aws-region#:~:text=I%20was%20looking%20for%20the%20same%20answer%20for%20a%20different%20problem.%20I%20wanted%20to%20get%20the%20region%20for%20a%20name%20of%20a%20role%2C%20I%20was%20able%20to%20get%20the%20info%20by%20doing%20this%3A

File: terraform/aws/analytical-platform-development/control-panel-message-broker/iam-policies.tf [hunk 2]
```diff
@@ -0,0 +1,83 @@
+resource "aws_sqs_queue_policy" "s3_queue_iam_policy" {
```
Comment (julialawrence) [inline terraform/aws/analytical-platform-development/control-panel-message-broker/iam-policies.tf:1]: It's way better to use the [policy document datasource](https://registry.terraform.io/providers/hashicorp/aws/2.61.0/docs/data-sources/iam_policy_document) than inlining policies like this:

File: terraform/aws/analytical-platform-development/control-panel-message-broker/iam-policies.tf [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+          "sqs:SendMessage"
+        ],
+      "Resource": "${aws_sqs_queue.s3_queue.arn}"
+    }
+  ]
+}
+POLICY
+}
+
+resource "aws_sqs_queue_policy" "iam_queue_iam_policy" {
```
Comment (julialawrence) [inline terraform/aws/analytical-platform-development/control-panel-message-broker/iam-policies.tf:6]: Same here

File: terraform/aws/analytical-platform-development/control-panel-message-broker/iam-policies.tf [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    }
+  ]
+}
+POLICY
+}
+
+resource "aws_sqs_queue_policy" "auth_queue_iam_policy" {
+  queue_url = aws_sqs_queue.auth_queue.id
+
+  policy = <<POLICY
```
Comment (julialawrence) [inline terraform/aws/analytical-platform-development/control-panel-message-broker/iam-policies.tf]: And here

**Review by julialawrence — CHANGES_REQUESTED** [review]
Some good code hygiene changes, looks good otherwise.

File: terraform/aws/analytical-platform-development/control-panel-message-broker/sqs.tf [hunk 5]
```diff
@@ -0,0 +1,20 @@
+resource "aws_sqs_queue" "s3_queue" {
+  #ts:skip=AWS.SQS.NetworkSecurity.High.0570
+  name = "s3_queue"
+
```
Comment (julialawrence) [inline terraform/aws/analytical-platform-development/control-panel-message-broker/sqs.tf]:
```suggestion
```

File: terraform/aws/analytical-platform-development/control-panel-message-broker/sqs.tf [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  tags = var.tags
+}
+
+resource "aws_sqs_queue" "iam_queue" {
+  #ts:skip=AWS.SQS.NetworkSecurity.High.0570
+  name = "iam_queue"
+
+  tags = var.tags
```
Comment (julialawrence) [inline terraform/aws/analytical-platform-development/control-panel-message-broker/sqs.tf]:
```suggestion
  tags = var.tags
```

File: terraform/aws/analytical-platform-development/control-panel-message-broker/sqs.tf [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+}
+
+resource "aws_sqs_queue" "auth_queue" {
+  #ts:skip=AWS.SQS.NetworkSecurity.High.0570
+  name = "auth_queue"
+
```
Comment (julialawrence) [inline terraform/aws/analytical-platform-development/control-panel-message-broker/sqs.tf]:
```suggestion
```

**Review by julialawrence — DISMISSED** [review]
Some more nitty requests.

File: terraform/aws/analytical-platform-development/control-panel-message-broker/sqs.tf (hunk 6 above)
Comment (jacobwoffenden) [reply terraform/aws/analytical-platform-development/control-panel-message-broker/sqs.tf]: this isn't required anymore, we set default tags with the provider 😎

**Review by jacobwoffenden — DISMISSED** [review]

**Review by jacobwoffenden — APPROVED** [review]

**Pull request merged**